crates/ruff_python_formatter/src/comments/visitor.rs

```diff
@@ -137,23 +143,20 @@ impl<'a> CommentsVisitor<'a> {
                 slice: self.source_code.slice(*comment_range),
             };
 
-            self.builder.add_comment(place_comment(
-                comment,
-                &Locator::new(self.source_code.as_str()),
-            ));
+            self.collector
+                .add_comment(comment, &Locator::new(self.source_code.as_str()));
 
             self.comment_ranges.next();
         }
 
         self.preceding_node = Some(node);
     }
-
-    fn finish(self) -> CommentsMap<'a> {
-        self.builder.finish()
-    }
 }
 
-impl<'ast> PreorderVisitor<'ast> for CommentsVisitor<'ast> {
+impl<'ast, 'b, Collector> PreorderVisitor<'ast> for CommentsVisitor<'ast, 'b, Collector>
+where
+    Collector: CommentsCollector<'ast>,
```

Comment:
Adding a generic type here means that Rust monomorphizes the whole visitor implementation (with all walk methods) for each collector. That's rather a lot of code, causing longer compile time and an increase in binary size (for something that's only needed in dev builds).

Reply:
I can also pass a `Box<dyn Fn(...) -> ...>

Reply:
Yeah. I'm not sure what the best approach is. Maybe we use `Box<dyn Fn(...)>` and/or hide it behind a feature flag (like a `debug` or `dev` flag).